streams/src/test/java/org/apache/kafka/streams/processor/internals/TaskManagerTest.java

```diff
@@ -157,26 +165,82 @@ public void shouldIdempotentlyUpdateSubscriptionFromActiveAssignment() {
     }
 
     @Test
-    public void shouldReturnOffsetsForAllCachedTaskIdsFromDirectory() throws IOException {
-        final File[] taskFolders = asList(testFolder.newFolder("0_1"),
-                                          testFolder.newFolder("0_2"),
-                                          testFolder.newFolder("0_3"),
-                                          testFolder.newFolder("1_1"),
-                                          testFolder.newFolder("dummy")).toArray(new File[0]);
+    public void shouldReportOffsetSumsForValidLockedTasks() throws IOException {
+        final File[] taskFolders = asList(/* SHOULD report offsets for the following cases: */
+            testFolder.newFolder("0_0"),     // active running task
+            testFolder.newFolder("0_1"),     // active non-running task
+            testFolder.newFolder("0_2"),     // standby task
+            testFolder.newFolder("1_0"),     // unowned (unlocked) task with valid checkpoint
+                                          /* should NOT report offsets for the following: */
+            testFolder.newFolder("1_1"),     // unowned (unlocked) task without checkpoint file
+            testFolder.newFolder("1_2"),     // owned/locked by another thread
+            testFolder.newFolder("dummy"))   // some random non-task dir
+                                       .toArray(new File[0]);
+
+        final Map<TopicPartition, Long> task01ChangelogOffsets = mkMap(
+            mkEntry(t1p0, 1L),
+            mkEntry(t1p1, 2L)
+        );
+        final Map<TopicPartition, Long> task02ChangelogOffsets = mkMap(
+            mkEntry(t1p0, 5L),
+            mkEntry(t1p1, 10L)
+        );
+        final Map<TopicPartition, Long> task10ChangelogOffsets = mkMap(
+            mkEntry(t1p0, 20L),
+            mkEntry(t1p1, 30L)
+        );
+
+        final Long task00OffsetSum = Task.LATEST_OFFSET;
+        final Long task01OffsetSum = 3L;
+        final Long task02OffsetSum = 15L;
+        final Long task10OffsetSum = 50L;
```

Comment:
I've gotten conflicting reviews on computing the expected value by code vs by hand. I think in this case it's more appropriate to compute it "by hand" rather than write a small method to compute it from the map, since computing offset sums from a changelog offset map is exactly what we're testing here (`TaskManager#computeOffsetSum`)